3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,23 +8,22 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
 
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      !firstName
+        ? reject(new Error("You didn't pass in a first name!"))
+        : resolve(`${firstName} Doe`);
+    }, 1000);
+  });
 };
 
-function main() {
-  getAnonName('John', console.log);
-}
+const main = () => {
+  getAnonName('Farshad')
+    .then((fullName) => console.log(fullName))
+    .catch((err) => console.error(err));
```

Comment:
We might not need console.error here because whenever our promise rejects, it will reject with `Error` object of javascript. That means, even we console.log it, it will already behave like console.error.
```suggestion
    .catch((err) => console.log(err));
```